scripts/zones/Attohwa_Chasm/mobs/Xolotl.lua

```diff
@@ -3,20 +3,90 @@
 --  Mob: Xolotl
 -----------------------------------
 require("scripts/globals/titles")
+local ID = require("scripts/zones/Attohwa_Chasm/IDs")
 -----------------------------------
 local entity = {}
 
 entity.onMobSpawn = function(mob)
-    mob:setRespawnTime(0, true)
+    mob:setLocalVar("xolotlDead", 0)
+end
+
+entity.onMobFight = function(mob,target)
+    local timeInterval = mob:getBattleTime() % 6
+
+    -- Spawn skeleton pets
+    for i = 1, 2 do
+        local child = GetMobByID(mob:getID() + i)
+        if child:isSpawned() then
+            if target and child:getCurrentAction() == xi.act.ROAMING then -- doing nothing, make share enmity
+                child:updateEnmity(target)
+            end
+        elseif mob:getCurrentAction() ~= 30 and mob:actionQueueEmpty() and timeInterval == (i-1)*3 then -- not spawned, not casting, not using an ability and should summon
```

Comment:
other devs, I summon thee. Loops inside onMobFight give me a migraine. One of y'all check it out and I'll trust you.